datafusion/physical-expr/src/expressions/negative.rs

```diff
@@ -118,13 +118,11 @@ pub fn negative(
     arg: Arc<dyn PhysicalExpr>,
     input_schema: &Schema,
 ) -> Result<Arc<dyn PhysicalExpr>> {
+    // TODO: the data type checker should be done in the logical plan phase
```

Comment:
If datafusion dose this in the preview stage or the logical plan generation stage, I think we don't need to check the child/input data type for the physical expr generation.

cc @alamb 
Is there user case that user generate the physical plan or physical expr directly without the logical plan or logical stage?

Reply:
I am not sure if anyone creates physical Exprs directly without a logical_plan Expr. 

However, I agree with this approach to throw an error if the physical types don't line up -- automatic coercion should be done at the logical level

Reply:
> I am not sure if anyone creates physical Exprs directly without a logical_plan Expr.
> 
> However, I agree with this approach to throw an error if the physical types don't line up -- automatic coercion should be done at the logical level

Got it. 
If the phy expr can be created directly without the type coercion or type checker in the logical phase, we need this checker in the creation of physical expr.

Reply:
> If the phy expr can be created directly without the type coercion or type checker in the logical phase, we need this checker in the creation of physical expr.

That is a good point. 

Maybe the larger question is if we are ok requiring that the logical optimizer passes are run on `LogicalPlan`s and `Expr` to successfully create `ExecutionPlan`s and `PhysicalExpr`s, 

Now that I type that it seems there is value in supporting creating `Exprs` and then PhysicalExprs directly (as part of implementing extensions, for example) 🤔 So maybe we should just ensure it is possible to call the coercion logic directly on a `Expr`